tools/wake/main.cpp

```diff
@@ -647,27 +717,29 @@ int main(int argc, char **argv) {
 
   TypeVar type = top->body->typeVar;
 
-  if (parse) top->format(std::cout, 0);
+  if (clo.parse) top->format(std::cout, 0);
   if (notype) return (ok && !terminalReporter.errors) ? 0 : 1;
 
   /* Setup logging streams */
-  if (noexecute && !fd1) fd1 = "error";
-  if (debug && !fd1) fd1 = "debug,info,echo,report,warning,error";
-  if (verbose && !fd1) fd1 = "info,echo,report,warning,error";
-  if (quiet && !fd1) fd1 = "error";
-  if (!tty && !fd1) fd1 = "echo,report,warning,error";
-  if (!fd1) fd1 = "report,warning,error";
-  if (!fd2) fd2 = "error";
-
-  status_set_bulk_fd(1, fd1);
-  status_set_bulk_fd(2, fd2);
-  status_set_bulk_fd(3, fd3);
-  status_set_bulk_fd(4, fd4);
-  status_set_bulk_fd(5, fd5);
+  if (noexecute && !clo.fd1) clo.fd1 = "error";
+  if (clo.debug && !clo.fd1) clo.fd1 = "debug,info,echo,report,warning,error";
+  if (clo.verbose && !clo.fd1) clo.fd1 = "info,echo,report,warning,error";
+  if (clo.quiet && !clo.fd1) clo.fd1 = "error";
+  if (!clo.tty && !clo.fd1) clo.fd1 = "echo,report,warning,error";
+  if (!clo.fd1) clo.fd1 = "report,warning,error";
+  if (!clo.fd2) clo.fd2 = "error";
```

Comment:
There's only a few places where we're editing the supposed "command-line" options so it's probably not worth setting up a separate stage for populating `clo` with default values, but this use of it as a mutable store (where it's less obvious it's just assigning defaults) feels a bit weird given the name.